go/workflow.go

```diff
@@ -0,0 +1,270 @@
+		ParentClosePolicy:     enums.PARENT_CLOSE_POLICY_ABANDON,
+		WorkflowIDReusePolicy: enums.WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE,
+	}
+	ctx = workflow.WithChildOptions(ctx, childWorkflowOptions)
+
+	child = workflow.ExecuteChildWorkflow(ctx, CustomerLoyaltyWorkflow, guest)
+	// wait for it to start before returning.
+	err = child.GetChildWorkflowExecution().Get(ctx, nil)
+	return err, child
+}
+
+func validateCustomerInfo(customer *CustomerInfo) {
+	customer.Status = status.NewStatus(customer.StatusLevel)
+	if len(customer.Guests) == 0 {
+		customer.Guests = make(map[string]struct{}, customer.Status.NumGuestsAllowed())
+	}
+}
+
+func signalAddPoints(ctx workflow.Context, c workflow.ReceiveChannel, customer *CustomerInfo) (err error) {
+	logger := workflow.GetLogger(ctx)
+	activities := Activities{}
+
+	var pointsToAdd int
+	c.Receive(ctx, &pointsToAdd)
+
+	logger.Info("Adding points to customer account.", "PointsAdded", pointsToAdd)
+	customer.LoyaltyPoints += pointsToAdd
+
+	statusChange := customer.Status.Update(customer.LoyaltyPoints)
+
+	if statusChange > 0 {
+		err = workflow.ExecuteActivity(ctx, activities.SendEmail,
+			fmt.Sprintf(common.EmailPromoted, customer.Status.Name())).
+			Get(ctx, nil)
+		if err != nil {
+			return errors.Wrap(err, "Error running SendEmail activity for status promotion.")
+		}
+	} else if statusChange < 0 {
+		err = workflow.ExecuteActivity(ctx, activities.SendEmail,
+			fmt.Sprintf(common.EmailDemoted, customer.Status.Name())).
+			Get(ctx, nil)
+		if err != nil {
+			return errors.Wrap(err, "Error running SendEmail activity for status demotion.")
+		}
+	}
+
+	return nil
+}
+
+func signalInviteGuest(ctx workflow.Context, c workflow.ReceiveChannel, customer *CustomerInfo) (err error) {
+	logger := workflow.GetLogger(ctx)
+	activities := Activities{}
+
+	var emailToSend string
+
+	if len(customer.Guests) < customer.Status.NumGuestsAllowed() {
+		var guestId string
+		c.Receive(ctx, &guestId)
+
+		logger.Info("Customer is allowed to invite guests. Attempting to invite.",
+			"GuestId", guestId)
+
+		guest := CustomerInfo{
+			CustomerId:    guestId,
+			AccountActive: true,
+		}
+
+		customer.Guests[guestId] = struct{}{}
+
+		err, guestWorkflow := startGuestWorkflow(ctx, guest)
+		logger.Info("Results from starting guest.", "error", err, "guest", guestWorkflow)
+
+		if _, ok := err.(*serviceerror.WorkflowExecutionAlreadyStarted); ok {
+			emailToSend = common.EmailGuestInvited
+			previousTier := customer.Status.PreviousTier()
+			err = guestWorkflow.SignalChildWorkflow(ctx, common.SignalEnsureMinimumStatus, previousTier).
+				Get(ctx, nil)
+			if _, ok := err.(*serviceerror.WorkflowExecutionAlreadyStarted); ok {
+				logger.Info("Failed to signal 'already started' guest account; child workflow likely closed.")
+				emailToSend = common.EmailGuestCanceled
+			} else if err != nil {
+				return errors.Wrapf(err, "Could not signal guest/child workflow for guest Id '%v'.", guestId)
```

Comment:
```suggestion
				return fmt.Errorf("could not signal guest/child workflow for guest Id '%v': %w", guestId, err)
```

This is the more common way to wrap errors (and Go doesn't want error phrases capitalized)